### DISCUSSION TITLE
Make "INVALID_FOREIGN_KEY" error non - fail on item creation if the referenced object does not (yet) exist

### DISCUSSION THREAD

**Discussion by helge000** [directus/directus#6428] [Archive]
Since a few versions back there where rather major changes to objects the API returns , namely relations and items.

Items now have a (redundant?) relation key for their foreign relation like this:
```yaml
 - id: 1
  other_collection_id: 1
# ...
```

This property makes my item dependent on the other item - creating a chicken in the egg problem for me. I also guess this property is generated and managed completely by directus.

When creating an item via API, this key needs to be removed - otherwise triggering:
```json
[{"message":"Invalid foreign key in field \"other_collection_id\".",
"extensions": {
    "code":"INVALID_FOREIGN_KEY",
    "collection":"other_collection_translations",
    "field":"other_collection_id","invalid":"1"
}}]
```
Since we also use this naming in many2many collection relation tables, it is nearly impossible to "guess" when these keys have to be removed or not from the item object. There, I have to post this field and it is perfectly valid to include non-existing items.

### Background
I am working on a cli to export and restore models, items, assets ect. Until rc-64 everything was working perfectly; with recent changes things are breaking down fast and leave a lot of guesswork for me.

**Comment by rijkvanzanten** [comment]
> Items now have a (redundant?) relation key for their foreign relation like this:

No fields where added in the foreign key constraint updates 🤔 This field must've been there before as well.

> Since we also use this naming in many2many collection relation tables, it is nearly impossible to "guess" when these keys have to be removed or not from the item object. 

Have you tried configuring the relationship to set the value to NULL on delete? It sounds like that should solve the problems you're running in to
